kanalony-storage-client/src/main/scala/kanalony/storage/generator/QueryMethodsGenerator.scala

```diff
@@ -0,0 +1,93 @@
+package kanalony.storage.generator
+
+/**
+ * Created by elad.benedict on 2/9/2016.
+ */
+
+class QueryMethodsGenerator(val tm : TableMetadata) {
+
+  private def getRangeParamNames(colDef : IColumnDefinition)  = {
+    (colDef.name + "Start", colDef.name + "End")
+  }
+
+  private def generateColDefs(cols : List[IColumnQueryDefinition]) : String = cols match {
+    case x::Nil => {
+      genParamDefinition(x)
+    }
+    case x::xs => {
+      genParamDefinition(x) + ", " + generateColDefs(xs)
+    }
+  }
+
+  private def genParamDefinition(cqd : IColumnQueryDefinition) = {
+    cqd.queryKind match {
+      case ColumnQueryKind.Equality => {
+        cqd.name + " : " + cqd.typeName
+      }
+      case ColumnQueryKind.Range => {
+        getRangeParamNames(cqd)._1 + " : " + cqd.typeName + ", " +
+        getRangeParamNames(cqd)._2 + " : " + cqd.typeName
+      }
+    }
+  }
+
+  private def generateWhereCondition(x: IColumnDefinition) = {
```

Comment:
Not sure if cd or cqd are better argument names than using x, but let's be consistent :)